packages/gnarly-reducer-erc20/src/reducer.ts

```diff
@@ -0,0 +1,110 @@
+import BN = require('bn.js')
+import makeDebug = require('debug')
+const debug = makeDebug('gnarly-reducer:erc20')
+
+import {
+  addABI,
+  Block,
+  getLogs,
+  IReducer,
+  ITypeStore,
+  ReducerType,
+  toHex,
+} from '@xlnt/gnarly-core'
+
+import abi from './abi'
+
+const TRANSFER_REASON = 'ERC20_TRANSFER'
+
+const makeReducer = (key: string, typeStore: ITypeStore) => (
+  darAddress: string,
+): IReducer => {
+  addABI(darAddress, abi)
+
+  interface IERC20Tracker {
+    balances: {
+      [id: string]: {
+        id: string,
+        darAddress: string;
+        owner: string;
+        balance: string;
+      },
+    },
+  }
+
+  const makeActions = (state: IERC20Tracker, { operation, emit }) => ({
+    transfer: (from: string, to: string, value: string) => {
+      debug('transferring %d tokens from %s to %s', value, from, to)
+
+      // turn value into BN value
+      value = new BN(value)
```

Comment:
let's use an immutable pattern here to avoid any confusion about the types

```js

transfer: (from: string, to: string, rawTokenValue: string) => {
   const tokenValue = new BN(rawTokenValue)
   ...
```